examples/src/keyrings/raw_rsa_keyring_example.py

```diff
@@ -0,0 +1,234 @@
+from aws_cryptographic_materialproviders.mpl.config import MaterialProvidersConfig
+from aws_cryptographic_materialproviders.mpl.models import CreateRawRsaKeyringInput, PaddingScheme
+from aws_cryptographic_materialproviders.mpl.references import IKeyring
+from aws_cryptographic_materialproviders.smithygenerated.aws_cryptography_materialproviders.errors import (
+    CollectionOfErrors,
+)
+from cryptography.hazmat.backends import default_backend as crypto_default_backend
+from cryptography.hazmat.primitives import serialization as crypto_serialization
+from cryptography.hazmat.primitives.asymmetric import rsa
+from typing import Dict
+
+import aws_encryption_sdk
+from aws_encryption_sdk import CommitmentPolicy
+
+# TODO-MPL: Remove this as part of removing PYTHONPATH hacks.
+MODULE_ROOT_DIR = '/'.join(__file__.split("/")[:-1])
+
+sys.path.append(MODULE_ROOT_DIR)
+
+EXAMPLE_DATA: bytes = b"Hello World"
+
+
+def should_generate_new_rsa_key_pair(public_key, private_key):
+    """Returns True if user doesn't provide keys, and we need to generate them and
+    returns False if the user has already provided both public and private keys
+    Raises an AssertionError if the user only provides one of private_key and public_key
+
+    Usage: should_generate_new_rsa_key_pair(public_key, private_key)
+    """
+    # If only one of public_key and private_key is provided, raise an Assertion Error
+    if (public_key and not private_key) or (not public_key and private_key):
+        raise AssertionError("Either both public and private keys should be provided! Or no keys \
+                             should be provided and the example can create the keys for you!")
+
+    # If no keys are provided, we should generate a new rsa key pair, so return True
+    if not public_key and not private_key:
+        return True
+
+    # If both keys are already provided, return False
+    return False
+
+
+def generate_rsa_keys():
+    """Generates a 4096-bit RSA public and private key pair
+
+    Usage: generate_rsa_keys()
+    """
+    ssh_rsa_exponent = 65537
+    bit_strength = 4096
+    key = rsa.generate_private_key(
+        backend=crypto_default_backend(),
+        public_exponent=ssh_rsa_exponent,
+        key_size=bit_strength
+    )
+
+    # This example choses a particular type of encoding, format and encryption_algorithm
+    # Users can choose the PublicFormat, PrivateFormat and encryption_algorithm that align most
+    # with their use-cases
+    public_key = key.public_key().public_bytes(
+        encoding=crypto_serialization.Encoding.PEM,
+        format=crypto_serialization.PublicFormat.SubjectPublicKeyInfo
+    )
+    private_key = key.private_bytes(
+        encoding=crypto_serialization.Encoding.PEM,
+        format=crypto_serialization.PrivateFormat.TraditionalOpenSSL,
+        encryption_algorithm=crypto_serialization.NoEncryption()
+    )
+
+    return public_key, private_key
+
+
+def create_rsa_keyring(public_key, private_key):
```

Comment:
I kind of want this to be 

```suggestion
def create_rsa_keyring(public_key_file_name, private_key_file_name):
```

like in https://github.com/aws/aws-encryption-sdk-dafny/blob/mainline/AwsEncryptionSDK/runtimes/net/Examples/Keyring/RawRSAKeyringExample.cs

Users interested in this keyring may bring their own PEM files, and it would be good to meet them where they are.

Reply:
Makes sense, good catch!